platform/src/syscalls_impl.rs

```diff
@@ -35,6 +35,19 @@ impl<S: RawSyscalls> Syscalls for S {
         }
     }
 
+    fn yield_wait_for(driver_number: u32, subscribe_number: u32) {
```

Comment:
```suggestion
    fn yield_wait_for(driver_number: u32, subscribe_number: u32) -> (u32, u32, u32) {
```

`yield-wait-for` returns values instead of calling an upcall, all registers return. This is similar to `yield-no-wait`, but it only returns `r1`.